support/reftest/reftest.py

```diff
@@ -0,0 +1,310 @@
+                "metadata": {"M": metadata},
+            },
+        )
+
+
+class S3Handler:
+    def __init__(
+        self,
+        bucket,
+        access_id=None,
+        access_key=None,
+        session_token=None,
+        region_name=None,
+    ):
+        self.s3 = boto3.Session(
+            aws_access_key_id=access_id,
+            aws_secret_access_key=access_key,
+            aws_session_token=session_token,
+            region_name=region_name,
+        ).client("s3")
+
+        self.bucket = bucket
+
+    def upload_from_localfile(self, path):
+        def _generate_content_type(path):
+            ext = os.path.splitext(path)[1]
+            if ext == ".xml":
+                return {"ContentType": "application/xml"}
+            if ext == ".gz":
+                return {"ContentType": "application/x-gzip"}
+            if ext == ".bz2":
+                return {"ContentType": "application/x-bzip"}
+
+            return {}
+
+        content_type = _generate_content_type(path)
+
+        def _generate_checksum(path):
+            if os.path.isfile(path):
+                sha256 = hashlib.sha256()
+                with open(path, "rb") as binary:
+                    while True:
+                        chunk = binary.read(4096)
+                        if not chunk:
+                            break
+                        sha256.update(chunk)
+
+                return sha256.hexdigest()
+
+            return None
+
+        checksum = _generate_checksum(path)
+
+        with open(path, "rb") as data:
+            self.s3.upload_fileobj(
+                Fileobj=data, Bucket=self.bucket, Key=checksum, ExtraArgs=content_type
+            )
+
+        return checksum
+
+    def delete_from_key(self, key):
+        return self.s3.delete_object(Bucket=self.bucket, Key=key)
+
+
+def parse_aws_info(parser):
+    parser.add_argument(
+        "--aws-access-id",
+        default=None,
+        help="Access ID for Amazon services. If no ID is provided, attempts to"
+        " find it among environment variables and ~/.aws/config file will"
+        " be made",
+    )
+    parser.add_argument(
+        "--aws-access-key",
+        default=None,
+        help="Access key for Amazon services. If no key is provided, attempts"
+        " to find it among environment variables and ~/.aws/config file"
+        " will be made",
+    )
+    parser.add_argument(
+        "--aws-session-token",
+        default=None,
+        help="Session token for Amazon services. If no token is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+    parser.add_argument(
+        "--default-region",
+        default=None,
+        help="Default region for Amamzon services. If no region is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+
+
+def parse_cdn_certs(parser):
+    parser.add_argument(
```

Comment:
For these arguments, there are some existing conventions which could be used for default values.  It would make the command a little easier to run.  Would you mind adding those defaults?

They are:

- cert: ~/certs/rcm-debug/rcm-debug.crt
- key: ~/certs/rcm-debug/rcm-debug.key
- cacert: /etc/rhsm/ca/redhat-uep.pem